src/journal_entry.rs

```diff
@@ -27,20 +29,32 @@ impl JournalEntry {
 		Self::from(fields)
 	}
 
-	pub fn get_field(&self, field: &str) -> Option<&str> {
-		self.fields.get(field).map(|v| v.as_str())
+	pub fn get_field_binary(&self, field: &str) -> Option<&[u8]> {
+		self.fields.get(field).map(|v| v.as_slice())
+	}
+
+	pub fn get_field_string(&self, field: &str) -> Result<Option<&str>, str::Utf8Error> {
+		self.fields
+			.get(field)
+			.map(|v| str::from_utf8(v))
+			.transpose()
```

Comment:
This transpose is weird to me. It seems weird that I have to check for the `Utf8Error` before even knowing if the field exists.